kernel/safe_typing.ml

```diff
@@ -319,6 +320,7 @@ type t = { seff : side_effect list; elts : SeffSet.t }
 
 let repr eff = eff.seff
 let empty = { seff = []; elts = SeffSet.empty }
+let is_empty { seff; elts } = List.is_empty seff && SeffSet.is_empty elts
```

Comment:
Not critical, but only one of the checks is needed, since the invariant guarantees that the list contains exactly the same elements as the set.

Reply:
@ejgallego not changing this?

Reply:
Nah it's trivial I didn't bother; anyways all this crazy stuff is already gone in my branch that makes entries private and thus properly handles side effects.